### ISSUE TITLE
ADFS 2019 update

### ISSUE THREAD

**Issue by GrahamJB1** [MicrosoftDocs/windowsserverdocs#3978]
The above did not work on ADFS 2019.

This is an updated script that:
1) sets the placeholder text to custom text (username)
2) updates the login field to prepend a default domain (contoso.com\)

Place the following at the bottom of onload.js

if (typeof Login != 'undefined'){  
    function UpdateLoginHandling() {
        var userName;
        if (typeof Login != 'undefined'){
            userName = document.getElementById(Login.userNameInput);
        }
        if (typeof UpdatePassword != 'undefined'){
            userName = document.getElementById(UpdatePassword.userNameInput);
        }
        if (typeof userName != 'undefined'){
            userName.setAttribute("placeholder","username");
            userName.type = "text";
            userName.setAttribute("onkeyup", "if (this.value && !this.value.match(/[@\\\\]/)) {this.value = 'contoso.com\\\\' + this.value } ");
        }
    }
 
    document.addEventListener("DOMContentLoaded", function(){
      // Handler when the DOM is fully loaded
      UpdateLoginHandling()
    });
}  

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 110e65f0-76bd-55d7-51b3-58149c8e0eb7
* Version Independent ID: 1184ab13-6803-236d-cc17-b08896cf1695
* Content: [Advanced Customization of AD FS Sign-in Pages](https://docs.microsoft.com/en-us/windows-server/identity/ad-fs/operations/advanced-customization-of-ad-fs-sign-in-pages#feedback)
* Content Source: [WindowsServerDocs/identity/ad-fs/operations/Advanced-Customization-of-AD-FS-Sign-in-Pages.md](https://github.com/MicrosoftDocs/windowsserverdocs/blob/master/WindowsServerDocs/identity/ad-fs/operations/Advanced-Customization-of-AD-FS-Sign-in-Pages.md)
* Product: **windows-server**
* Technology: **identity-adfs**
* GitHub Login: @billmath
* Microsoft Alias: **billmath**

**Comment by susmitkoirala** [comment]
I have tried to overwrite the submitLoginRequest function on my AD FS 3.0 hosted on a windows server 2016 and I am getting an error. Also tried the proposed solution by @GrahamJB1 but no luck. Can you please let me know if there is a fix in the pipeline for this issue?

**Comment by SamuelBowman** [comment]
I haven't tested on Server 2016 (only 2019), but I believe this issue is connected to whether or not the new paginated style is being used. Instead of validating usernames before submitting the login request, the paginated style validates usernames before moving on to the next screen. For this reason, validateAndNext must be overridden instead of submitLoginRequest. I initially wasted about an hour troubleshooting this issue and eventually writing my own code to implement this before I found #2737 which does the same thing with added backwards compatibility for the old non-paginated style. For some reason, despite being an issue referencing the same page of documentation, it isn't linked to by docs.microsoft.com, hence why I'm writing this comment.

TL;DR use the code in #2737

**Comment by brightjdb** [comment]
> I haven't tested on Server 2016 (only 2019), but I believe this issue is connected to whether or not the new paginated style is being used. Instead of validating usernames before submitting the login request, the paginated style validates usernames before moving on to the next screen. For this reason, validateAndNext must be overridden instead of submitLoginRequest. I initially wasted about an hour troubleshooting this issue and eventually writing my own code to implement this before I found #2737 which does the same thing with added backwards compatibility for the old non-paginated style. For some reason, despite being an issue referencing the same page of documentation, it isn't linked to by docs.microsoft.com, hence why I'm writing this comment.
> 
> TL;DR use the code in #2737

Samuel,

I'm having a hard time getting this code to work.  Not sure if it's a newer version of the onload.js  Could I send some of the code to you or help me understanding where I need to place this?  I have it working without pagination

**Comment by billmath** [comment]
Thank you for submitting this issue.  Unfortunately, due to high volumes I was unable to respond to this in a timely manner.  At this point, I am going to close this due to it's age and the possibility that it may have been addressed.  If this is still a problem, please consider submitting a new issue.  Thank you.

#please-close
